libs/partners/google-vertexai/langchain_google_vertexai/llms.py

```diff
@@ -0,0 +1,525 @@
+from __future__ import annotations
+
+from concurrent.futures import Executor, ThreadPoolExecutor
+from typing import TYPE_CHECKING, Any, ClassVar, Dict, Iterator, List, Optional, Union
+
+from langchain_core.callbacks.manager import (
+    AsyncCallbackManagerForLLMRun,
+    CallbackManagerForLLMRun,
+)
+from langchain_core.language_models.llms import BaseLLM
+from langchain_core.outputs import Generation, GenerationChunk, LLMResult
+from langchain_core.pydantic_v1 import BaseModel, Field, root_validator
+
+from langchain_google_vertexai.utils import (
+    create_retry_decorator,
+    get_client_info,
+    init_vertexai,
+    raise_vertex_import_error,
+)
+
+if TYPE_CHECKING:
+    from google.cloud.aiplatform.gapic import (
```

Comment:
if we want, we can make this required now

Reply:
done